dozer-types/src/types/mod.rs

```diff
@@ -331,3 +340,91 @@ pub enum Operation {
     Insert { new: Record },
     Update { old: Record, new: Record },
 }
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct DozerPoint(pub Point<OrderedFloat<f64>>);
+
+impl GeodesicDistance<OrderedFloat<f64>> for DozerPoint {
+    fn geodesic_distance(&self, rhs: &Self) -> OrderedFloat<f64> {
+        let f = point! { x: self.0.x().0, y: self.0.y().0 };
+        let t = point! { x: rhs.0.x().0, y: rhs.0.y().0 };
+        OrderedFloat(f.geodesic_distance(&t))
+    }
+}
+
+impl Ord for DozerPoint {
+    fn cmp(&self, other: &Self) -> Ordering {
+        if self.0.x() == other.0.x() && self.0.y() == other.0.y() {
+            Ordering::Equal
+        } else if self.0.x() > other.0.x()
+            || (self.0.x() == other.0.x() && self.0.y() > other.0.y())
+        {
+            Ordering::Greater
+        } else {
+            Ordering::Less
+        }
+    }
+}
+
+impl PartialOrd for DozerPoint {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl FromStr for DozerPoint {
+    type Err = TypeError;
+
+    fn from_str(str: &str) -> Result<Self, Self::Err> {
+        let s = str.replace('(', "");
+        let s = s.replace(')', "");
+        let mut cs = s.split(',');
+        let x = cs
+            .next()
+            .ok_or(InvalidFieldValue(s.clone()))?
+            .parse::<f64>()
+            .map_err(|_| InvalidFieldValue(s.clone()))?;
+        let y = cs
+            .next()
+            .ok_or(InvalidFieldValue(s.clone()))?
+            .parse::<f64>()
+            .map_err(|_| InvalidFieldValue(s.clone()))?;
+        Ok(Self(Point::from((OrderedFloat(x), OrderedFloat(y)))))
+    }
+}
+
+impl From<(f64, f64)> for DozerPoint {
+    fn from((x, y): (f64, f64)) -> Self {
+        Self(point! {x: OrderedFloat(x), y: OrderedFloat(y)})
+    }
+}
+
+impl Display for DozerPoint {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&format!("{:?}", self.0.x_y()))
+    }
+}
+
+impl DozerPoint {
+    pub fn to_bytes(&self) -> [u8; 16] {
+        let mut result = [0_u8; 16];
+        result[0..8].copy_from_slice(&self.0.x().to_be_bytes());
+        result[8..16].copy_from_slice(&self.0.y().to_be_bytes());
+        result
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> Result<(f64, f64), DeserializationError> {
+        let x = f64::from_be_bytes(
+            bytes[0..7]
```

Comment:
Should be `[0..8]`? And is it better to just return `Self`? Like `pub fn from_bytes(bytes: [u8; 16]) -> Self` and let the length error be handled elsewhere?

Reply:
Thanks. Fixed.